src/java/mapImageWriter/MapImageWriter.java

```diff
@@ -0,0 +1,190 @@
+            writeLayer1 = true;
+            writeLayer2 = true;
+            writeLayer3 = true;
+            writeLayer4 = true;
+            writeLayer5 = true;
+            writeLayer6 = true;
+            writeLayer7 = true;
+            writeLayer8 = true;
+            writeLayerh = true;
+        }
+    }
+
+    public void importMap() {
+        try {
+            File dir = inMapPath.toFile();
+
+            File[] mapFiles = dir.listFiles((dir1, filename) -> filename.endsWith(".scmap"));
+            if (mapFiles == null || mapFiles.length == 0) {
+                System.out.println("No scmap file in map folder");
+                return;
+            }
+            map = SCMapImporter.loadSCMAP(inMapPath);
+            SaveImporter.importSave(inMapPath, map);
+        } catch (IOException e) {
+            e.printStackTrace();
+            System.err.println("Error while importing the map.");
+        }
+    }
+
+    public void writeMapImages() throws IOException {
+
+        Random random = new Random();
+        FloatMask heightmapBase = map.getHeightMask(symmetrySettings);
+        heightmapBase = new FloatMask(heightmapBase, random.nextLong());
+        heightmapBase.applySymmetry(SymmetryType.SPAWN);
+        map.setHeightImage(heightmapBase);
+
+        FloatMask[] texturesMasks = map.getTextureMasksScaled(symmetrySettings);
+        FloatMask oldLayer1 = texturesMasks[0];
+        FloatMask oldLayer2 = texturesMasks[1];
+        FloatMask oldLayer3 = texturesMasks[2];
+        FloatMask oldLayer4 = texturesMasks[3];
+        FloatMask oldLayer5 = texturesMasks[4];
+        FloatMask oldLayer6 = texturesMasks[5];
+        FloatMask oldLayer7 = texturesMasks[6];
+        FloatMask oldLayer8 = texturesMasks[7];
+        if(writeLayerh) {
+            util.ImageUtils.writePNGFromMask(heightmapBase, 1, writeImagesPath + "\\Heightmap.png");
+        }
+        if(writeLayer1) {
+            util.ImageUtils.writePNGFromMask(oldLayer1, 255, writeImagesPath + "\\Layer 1.png");
+        }
+        if(writeLayer2) {
+            util.ImageUtils.writePNGFromMask(oldLayer2, 255, writeImagesPath + "\\Layer 2.png");
+        }
+        if(writeLayer3) {
+            util.ImageUtils.writePNGFromMask(oldLayer3, 255, writeImagesPath + "\\Layer 3.png");
+        }
+        if(writeLayer4) {
+            util.ImageUtils.writePNGFromMask(oldLayer4, 255, writeImagesPath + "\\Layer 4.png");
+        }
+        if(writeLayer5) {
+            util.ImageUtils.writePNGFromMask(oldLayer5, 255, writeImagesPath + "\\Layer 5.png");
+        }
+        if(writeLayer6) {
+            util.ImageUtils.writePNGFromMask(oldLayer6, 255, writeImagesPath + "\\Layer 6.png");
+        }
+        if(writeLayer7) {
+            util.ImageUtils.writePNGFromMask(oldLayer7, 255, writeImagesPath + "\\Layer 7.png");
+        }
+        if(writeLayer8) {
+            util.ImageUtils.writePNGFromMask(oldLayer8, 255, writeImagesPath + "\\Layer 8.png");
+        }
```

Comment:
make writeImagesPath a Path object and use resolve for better cross platform

Reply:
I used Paths.get() instead